packages/yew-macro/src/html_tree/html_component.rs

```diff
@@ -141,29 +141,52 @@ impl HtmlComponent {
         Some(cursor)
     }
 
-    fn path_arguments(cursor: Cursor) -> Option<(PathArguments, Cursor)> {
-        let (punct, cursor) = cursor.punct()?;
+    /// Refer to the [`syn::parse::Parse`] implementation for [`AngleBracketedGenericArguments`].
+    fn path_arguments(mut cursor: Cursor) -> Option<(PathArguments, Cursor)> {
```

Comment:
Hmm... I still really want to get rid of this and the `peek_type` method. Did you run into an issue here?
Other than that this PR is pretty much done, should we just merge this and revisit this topic another time? I don't want to push you to do things you don't really want to do.

Reply:
The `Peek` implementation depends on `peek_type` which dependends on `path_arguments`. I think it would be best if this were merged first and I'll try to work on removing this in another PR.

Reply:
alright, sounds like a plan